frontend/src/app/map/interactive-places/interactive-places.component.html

```diff
@@ -0,0 +1,551 @@
+<svg>
+  <g id="clickable-wcs" class="clickableClass">
+    <rect id="wc1" name="wc" x="354" y="409" class="clickableWcClass" width="107" height="83" />
+    <rect id="wc2" name="wc" x="504" y="409" class="clickableWcClass" width="109" height="83" />
+    <rect id="wc3" name="wc" x="354" y="885" class="clickableWcClass" width="68" height="61" />
+    <rect id="wc4" name="wc" x="422" y="991" class="clickableWcClass" width="77" height="121" />
+    <rect id="wc5" name="wc" x="540" y="992" class="clickableWcClass" width="73" height="120" />
+    <rect id="wc6" name="wc" x="1693" y="405" class="clickableWcClass" width="108" height="87" />
+    <rect id="wc7" name="wc" x="1843" y="405" class="clickableWcClass" width="63" height="128" />
+    <rect id="wc8" name="wc" x="1884" y="886" class="clickableWcClass" width="69" height="60" />
+    <rect id="wc9" name="wc" x="1693" y="989" class="clickableWcClass" width="73" height="124" />
+    <rect id="wc10" name="wc" x="1808" y="990" class="clickableWcClass" width="76" height="123" />
+  </g>
+  <g id="clickable-elevators" class="clickableClass">
+    <rect id="elevator1" name="elevator" x="479" y="719" class="clickableElevatorClass" width="64" height="165" />
+    <rect id="elevator2" name="elevator" x="1763" y="718" class="clickableElevatorClass" width="64" height="168" />
+  </g>
+  <g id="water_point">
+    <g id="water_point1" class="water" transform="translate(570,361), scale(0.07,0.07)">
+      <path d="M307.836,97.02C293.933,74.26,246.108,0,214.425,0c-6.917,0-12.838,3.265-16.673,9.192
+  c-12.997,20.085-32.308,42.406-52.754,66.037c-46.691,53.965-99.611,115.13-99.611,176.305
+  c0,88.644,72.117,160.761,160.761,160.761s160.761-72.117,160.761-160.761C366.908,197.892,321.694,119.706,307.836,97.02z
+   M206.147,392.295c-77.616,0-140.761-63.145-140.761-140.761c0-53.724,50.331-111.896,94.736-163.219
+  c20.914-24.172,40.668-47.004,54.42-68.257c0.013-0.019,0.024-0.037,0.036-0.054c3.41,0.152,13.381,5.282,30.024,23.733
+  c14.483,16.058,30.879,38.683,46.166,63.708c34.628,56.688,56.14,111.899,56.14,144.088
+  C346.908,329.15,283.763,392.295,206.147,392.295z" />
+      <path d="M274.414,312.717c-18.91,15.374-42.774,23.84-67.198,23.84c-23.371,0-45.554-7.418-64.151-21.454
+  c-4.409-3.327-10.679-2.451-14.006,1.958c-3.327,4.409-2.45,10.679,1.958,14.006c22.096,16.676,48.446,25.49,76.199,25.49
+  c29.004,0,57.35-10.058,79.814-28.322c4.286-3.484,4.935-9.782,1.451-14.067S278.699,309.233,274.414,312.717z" />
+    </g>
```

Comment:
Ten fragment jest bardzo słaby bo przez powtórzenia robi nam się duży html + tracimy szanse na proste modyfikowanie wszystkich elementów.
Lepiej zrobić sobie definicję:
``` html
  <defs>
    <g id="water-point-icon" transform="scale(0.07,0.07)">
      <path
        d="M307.836,97.02C293.933,74.26,246.108,0,214.425,0c-6.917,0-12.838,3.265-16.673,9.192
          c-12.997,20.085-32.308,42.406-52.754,66.037c-46.691,53.965-99.611,115.13-99.611,176.305
          c0,88.644,72.117,160.761,160.761,160.761s160.761-72.117,160.761-160.761C366.908,197.892,321.694,119.706,307.836,97.02z
           M206.147,392.295c-77.616,0-140.761-63.145-140.761-140.761c0-53.724,50.331-111.896,94.736-163.219
          c20.914-24.172,40.668-47.004,54.42-68.257c0.013-0.019,0.024-0.037,0.036-0.054c3.41,0.152,13.381,5.282,30.024,23.733
          c14.483,16.058,30.879,38.683,46.166,63.708c34.628,56.688,56.14,111.899,56.14,144.088
          C346.908,329.15,283.763,392.295,206.147,392.295z"
      />
      <path
        d="M274.414,312.717c-18.91,15.374-42.774,23.84-67.198,23.84c-23.371,0-45.554-7.418-64.151-21.454
    c-4.409-3.327-10.679-2.451-14.006,1.958c-3.327,4.409-2.45,10.679,1.958,14.006c22.096,16.676,48.446,25.49,76.199,25.49
    c29.004,0,57.35-10.058,79.814-28.322c4.286-3.484,4.935-9.782,1.451-14.067S278.699,309.233,274.414,312.717z"
      />
    </g>
  </defs>
```
I potem 
``` html
  <use xlink:href="#water-point-icon" id="water_point1" class="water" transform="translate(570,361)"></use>
  <use xlink:href="#water-point-icon" id="water_point2" class="water" transform="translate(325, 455)"></use>
```
jeżeli chodzi o konwencje nazywania klas zazwyczaj przyjmuje się kebab case czyli id zamiast `water_point1` to `water-point-1`
Generalnie jak widzicie, że musicie kopiować coś 1:1 to zazwyczaj da się to zrobić lepiej :) Jeżeli nie macie pomysłu warto poszukać rozwiązania na necie.

Żródło: `http://frontendbabel.info/articles/svg-grouping-and-reusing-elements/`